app/src/main/java/com/github/se/orator/model/profile/UserProfileRepositoryFirestore.kt

```diff
@@ -251,19 +251,7 @@ class UserProfileRepositoryFirestore(private val db: FirebaseFirestore) : UserPr
             // Extract 'recentData' list
             val recentDataList =
                 (statisticsMap["recentData"] as? List<Map<String, Any>>) ?: emptyList()
-            val recentData =
-                recentDataList.map { dataMap ->
-                  AnalysisData(
-                      transcription = dataMap["transcription"] as? String ?: "",
-                      fillerWordsCount = (dataMap["fillerWordsCount"] as? Number)?.toInt() ?: -1,
-                      averagePauseDuration =
-                          (dataMap["averagePauseDuration"] as? Number)?.toDouble() ?: -1.0,
-                      sentimentScore = (dataMap["sentimentScore"] as? Number)?.toDouble() ?: 0.0,
-                      talkTimePercentage =
-                          (dataMap["talkTimePercentage"] as? Number)?.toDouble() ?: -1.0,
-                      talkTimeSeconds = (dataMap["talkTimeSeconds"] as? Number)?.toDouble() ?: -1.0,
-                      pace = (dataMap["pace"] as? Number)?.toInt() ?: -1)
-                }
+            val recentData = recentDataList.map { dataMap -> mapToAnalysisData(dataMap) }
```

Comment:
Thanks for implementing what I suggested

Reply:
You are welcome.